Testscripts/Linux/validate-da.sh

```diff
@@ -220,11 +224,15 @@ function enable_disable_da(){
         fail_test "/dev/msda does not exist."
     fi
     # Check for "driver setup status=0" and "starting the dependency agent" in service.log.1 and service.log respectively
-    if ! grep -iq "driver setup status=0" $da_log_dir/service.log.1; then
+    if ! driver_status=$(sed -n 's/.*Driver setup status=//p' $da_log_dir/service.log.1); then
         fail_test "driver setup status=0 not found"
```

Comment:
This message is inaccurate.  `sed` will have a non-zero exit code if the script is malformed or an input file is unreadable.

I made a mistake when we talked on Friday.  The suggestion to replace `grep` with `sed` wasn't completely thought out. `grep` encodes information about whether the expression matched in its exit code. `sed` does not. `sed` will be useful, but we'll need to handle the output a little differently to differentiate between the line missing, a bad status, and the expected case of status=0.

Suggestion:

```
# Separating the line selection pattern and the transformation pattern will make it easier
# to differentiate between not matching the line versus getting an unexpected status.
if ! driver_status=$(sed -n '/Driver setup status=//s/^[^=]*//p' $da_log_dir/service.log.1); then
    fail_test ...
fi

case "$driver_status" in
    =0)
        # expected result
        ;;
    =*)
        # anything except =0 indicated setup failure
        fail_test Unexpected driver setup status $driver_status
        ;;
    *)
        fail_test did not find driver setup status
        ;;
esac
```

Reply:
Done.